tex-exchange-contract/src/main/kotlin/com/ccc/state/Stock.kt

```diff
@@ -0,0 +1,47 @@
+package com.ccc.state
+
+import com.ccc.contract.StockContract
+import net.corda.core.contracts.BelongsToContract
+import net.corda.core.contracts.LinearState
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+
+/**
+ * A [LinearState] describing a Stock.
+ * @param description Textual description of the  item.
+ * @param owner The Party who owns the  item.
+ * @param listed Whether or not the  item is listed in an active sale.
+ * @param linearId Unique identifier of a StockState object.
+ */
+@BelongsToContract(StockContract::class)
+data class Stock
+    (
+    override val linearId: UniqueIdentifier = UniqueIdentifier(),
+    val description: String,
+    val owner: Party,
+    val listed: Boolean = false
```

Comment:
Is this property necessary? It might make more sense to manage `listed`-ness with the creation / deletion of sell order states?

Reply:
Yes, indeed.  The listed property makes it convenient for queries because we don't have to join with Order.